### PR TITLE
GT-1043 Update tool UI controllers to be created via Dagger

### PR DESCRIPTION
**Pull request by frett**
This updates all the tool UI controllers to utilize Dagger for creation. The long term goal of this is to allow the controllers to be split between the various renderer modules based on where they are actually used

### CODE REVIEW

**Comment by frett** [comment]
ugg, Dagger isn't handling the multibind at the moment. I had tested it earlier and it was working :(

**Comment by frett** [comment]
Dagger 2.31 introduces native assisted injection, so I'm going to switch to that before this PR gets merged

File: ui/tract-renderer/build.gradle [hunk 1]
```diff
@@ -65,9 +65,12 @@ dependencies {
     implementation "com.tinder.statemachine:statemachine:${deps.tinderStateMachine}"
     implementation "jp.wasabeef:picasso-transformations:${deps.picassoTransforms}"
 
+    compileOnly "com.squareup.inject:assisted-inject-annotations-dagger2:${deps.assistedInject}"
```
Comment (frett) [inline ui/tract-renderer/build.gradle:68]: using square assisted inject is temporary until https://github.com/google/dagger/issues/2278 is released

**Comment by frett** [comment]
@gyasistory this is ready to review now

**Review by gyasistory — APPROVED** [review]

**Pull request merged**